nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java

```diff
@@ -70,11 +61,31 @@
 
     private static final Log log = LogFactory.getLog(PropertyCommentManager.class);
 
+    protected static final String GET_COMMENT_PAGEPROVIDER_NAME = "GET_COMMENT";
+
     protected static final String GET_COMMENTS_FOR_DOC_PAGEPROVIDER_NAME = "GET_COMMENTS_FOR_DOCUMENT";
 
+    protected static final String HIDDEN_FOLDER_TYPE = "HiddenFolder";
+
+    protected static final String COMMENT_NAME = "comment";
+
     @Override
     public List<DocumentModel> getComments(DocumentModel docModel) {
-        throw new UnsupportedOperationException();
+        try (CloseableCoreSession session = CoreInstance.openCoreSession(docModel.getRepositoryName())) {
+            return getComments(session, docModel);
```

Comment:
You should detach the DocumentModels you acquire from a privileged session and return outside of it, otherwise lazy fetch of non-prefetched schemas will fail — but I see lots of other code doing it in CommentManagerImpl so maybe a future cleanup...